Orange/canvas/report/report.py

```diff
@@ -263,17 +279,10 @@ def report_list(data,
             ) for rowi, row in enumerate(data))
 
         self.report_name(name)
-        if isinstance(table, QAbstractItemModel):
-            model = table
-        else:
-            try:
-                model = table.model()
-            except:
-                model = None
-        if isinstance(model, QStandardItemModel):
-            body = report_standard_model(model)
-        elif isinstance(model, QAbstractItemModel):
-            body = report_abstract_model(model, table)
+        if isinstance(table, QAbstractItemView):
+            body = report_abstract_model(table.model(), table)
+        elif isinstance(table, QAbstractItemModel):
+            body = report_abstract_model(table)
```

Comment:
This is not exactly the same: the previous code would take `table.model()` disregarding the type. But I agree that checking the type is better than blindly accepting (any) `table.model()`. I suppose that the previous code was a result of (non-)linear evolution of the code. :)